src/Raven.Server/RavenServer.cs

```diff
@@ -513,14 +513,37 @@ private void StartMonitoringCpuCredits()
             var sw = Stopwatch.StartNew();
             Stopwatch err = null;
 
+            int currentRetry = 0;
             try
             {
-                UpdateCpuCreditsFromExec();
+                // This is the first sync after a restart, so we want to retry until we get a valid result. Next sync is in 30 minutes.
+                while (true)
+                {
+                    try
+                    {
+                        UpdateCpuCreditsFromExec();
+                        break;
+                    }
+                    catch (Exception ex)
+                    {
+                        currentRetry++;
+
+                        // If we weren't able to sync at startup, we'll assume we have enough credits and let RavenDB perform in the meantime.
+                        // We don't want RavenDB to (false) throttle unless it knows for sure that it doesn't have credits.
+                        CpuCreditsBalance.RemainingCpuCredits = CpuCreditsBalance.MaxCredits;
+                        
+                        if (currentRetry > 5)
+                        {
+                            throw;
```

Comment:
What will happen at this point?

Reply:
Nothing will happen. It will be caught in the catch block, logged, and ignored. Then we'll try in the next time we have to sync (30 min)

Reply:
But at this point, we have 5 minutes that went by with no visible action, right?

Reply:
Ok. I merged it with the existing loop.